wpilibj/src/main/java/edu/wpi/first/wpilibj/Notifier.java

```diff
@@ -99,6 +99,18 @@ public Notifier(Runnable run) {
       }
     });
     m_thread.setDaemon(true);
+    m_thread.setUncaughtExceptionHandler((thread, error) -> {
+      Throwable cause = error.getCause();
+      if (cause != null) {
+        error = cause;
+      }
+      DriverStation.reportError("Unhandled exception: " + error.toString(), error.getStackTrace());
+      DriverStation.reportWarning("Robots should not quit, but yours did!", false);
+      DriverStation.reportError(
+          "The loopFunc() method (or methods called by it) should have handled "
+              + "the exception above.", false);
+      System.exit(1);
```

Comment:
Exiting here seems like a really bad approach. You're assuming that Notifier isn't used by anything else. I don't think a crash in a random thread is a good reason to kill the entire robot program unless the user asks for it.

Instead, the TimedRobot main loop should join the notifier thread.

Reply:
Aborting is the behavior for C++ threads already, and silently failing isn't desirable.

Reply:
Showing an error message is good. Exiting for all notifiers seems bad.